go/kbfs/libfs/leveldb_storage.go

```diff
@@ -255,7 +267,19 @@ func (fs *levelDBStorage) sync() (err error) {
 	return f.Lock()
 }
 
-func (fs *levelDBStorage) setMeta(fd storage.FileDesc) error {
+func (fs *levelDBStorage) sync() (err error) {
+	fs.syncLock.Lock()
+	defer fs.syncLock.Unlock()
+	return fs.syncLocked()
+}
+
+func (fs *levelDBStorage) syncRLocked() (err error) {
+	fs.syncLock.RUnlock()
+	defer fs.syncLock.RLock()
```

Comment:
The name of the method implies that when it's called it should already be locked while in fact it's locking inside. Maybe rename it?

Reply:
Same for `doLogRLocked` above. Perhaps `rlockAndSync` and `rlockAndDoLog`?

Reply:
@songgao I think you have it backwards.  This code is actually _unlocking_  and _re-locking_ it, not locking it.  It assumes it's already read-locked before called, as the name implies.

Reply:
I don't follow the point about `doLogRLocked()` -- that function doesn't do any locking and assumes the lock is already held.

Reply:
Ah you're right. I mis-read the `doLogRLocked` one because of the GH collapsing.